classes/editorialTask/Template.php

```diff
@@ -172,4 +167,67 @@ public function promote(Submission $submission): Task
         ]);
     }
 
+    /**
+     * Scope: filter by  type
+     */
+    public function scopeFilterByType(Builder $q, int $type): Builder
+    {
+        return $q->where('type', $type);
+    }
+
+    /**
+     * Scope: filter by title LIKE
+     */
+    public function scopeFilterByTitleLike(Builder $query, string $title): Builder
+    {
+        $title = trim($title);
+        if ($title === '') {
+            return $query;
+        }
+        $needle = '%' . str_replace(['%', '_'], ['\\%', '\\_'], mb_strtolower($title)) . '%';
+        return $query->whereRaw('LOWER(title) LIKE ?', [$needle]);
+    }
+
+    /**
+     * free-text/ words search across:
+     * title column
+     * name, description
+     */
+    public function scopeFilterBySearch(Builder $query, string $phrase): Builder
+    {
+        $phrase = trim($phrase);
+        if ($phrase === '') {
+            return $query;
+        }
+
+        $tokens = preg_split('/\s+/', $phrase) ?: [];
+        $tokens = array_values(array_filter($tokens, fn ($t) => $t !== ''));
```

Comment:
We also should do a special treatment for the task or discussion keyword within search phrase. E.g., the search phrase
```
task ethics
```
The code should perform the search by the keyword `ethics` within the tasks only.